### PR TITLE
Profiles tweaks

### PR DESCRIPTION
**Pull request by mattias-p**
I've described some details about most of the changes in the respective commit messages.

I took the liberty to rename load() to from_json(). I think it's more fitting.

### CODE REVIEW

File: lib/Zonemaster/Engine/Profile.pm [hunk 1]
```diff
@@ -166,21 +161,21 @@ profile JSON file.
 
 =head2 resolver.defaults.usevc
 
-A boolean. If C<true>, only use TCP. Default C<false>.
+1 or 0. If C<1>, only use TCP. Default C<0>.
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Profile.pm]: Why change from true/false to 1/0?

File: lib/Zonemaster/Engine/Profile.pm [hunk 2]
```diff
@@ -166,21 +161,21 @@ profile JSON file.
...
 
 =head2 resolver.defaults.retrans
 
-A number. The number of seconds between retries. Default 3.
+A non-negative integer. The number of seconds between retries. Default 3.
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Profile.pm]: Maybe a positive integer? Should zero be permitted?

**Comment by matsduf** [comment]
@vlevigneron, I think the idea was that there should be a helper method to load from file to JSON string, and then use "from_json". That's an extra step i e.g. CLI, but simple to implement.

The helper method should verify that the result is valid JSON, but not necessarily valid policy.

File: lib/Zonemaster/Engine/Profile.pm (hunk 2 above)
Comment (matsduf) [reply lib/Zonemaster/Engine/Profile.pm]: An integer between (and including) 1 and 240 is reasonable. Default 3 is fine.

File: lib/Zonemaster/Engine/Profile.pm [hunk 3]
```diff
@@ -166,21 +161,21 @@ profile JSON file.
...
 
 =head2 resolver.defaults.recurse
 
-A boolean. If C<true>, sets the RD flag in queries. Default C<false>.
+1 or 0. If C<1>, sets the RD flag in queries. Default C<0>.
 
-This should almost certainly be kept C<false>.
+This should almost certainly be kept C<0>.
 
 =head2 resolver.defaults.retry
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Profile.pm]: The resolver.defaults.retry should be a positive integer between (and including) 1 and 100. The default of 2 is fine.

File: lib/Zonemaster/Engine/Profile.pm (hunk 1 above)
Comment (mattias-p) [reply lib/Zonemaster/Engine/Profile.pm]: I was trying to take a shortcut here but I failed to consider the consequences with regard to JSON. I'm making an update that takes both Perl and JSON representations into consideration.

File: lib/Zonemaster/Engine/Profile.pm (hunk 2 above)
Comment (mattias-p) [reply lib/Zonemaster/Engine/Profile.pm]: Updating to positive integer.

**Comment by mattias-p** [comment]
> The "load_policy_file" method should also be specified.

I was assuming that the load_policy_file() helper should live in Zonemaster::Engine::Util.

File: lib/Zonemaster/Engine/Profile.pm (hunk 3 above)
Comment (mattias-p) [reply lib/Zonemaster/Engine/Profile.pm]: Updating to integer between 1 and 255. LDNS uses uint8 and zero doesn't make sense.

File: lib/Zonemaster/Engine/Profile.pm [hunk 4]
```diff
@@ -336,22 +334,19 @@ level for the tag.
 
 =head2 test_cases
 
-A hashref mapping test case names to booleans. Default C<{}>.
+An arrayref of names of test cases as listed in the L<test case specifications|
+https://github.com/zonemaster/zonemaster/tree/master/docs/specifications/tests>.
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Profile.pm]: "...as listed in the L<test case list|
https://github.com/zonemaster/zonemaster/tree/master/docs/specifications/tests/TestCases.md>"

File to be created before we release, see zonemaster/zonemaster#649

Comment (matsduf) [reply lib/Zonemaster/Engine/Profile.pm]: I will create a separte PR on this change.

**Pull request merged**